### PR TITLE
[Health API] Add 1-min, 5-min backpressure and multipipeline test cases.

### PR DESCRIPTION
**Pull request by mashhurs**
## Release notes
[rn:skip]

## What does this PR do?
Adds 1min, 5min and multi-pipeline integration tests for the health API

## Why is it important/What is the impact to the user?
- Validates further iteration changes.
- No impact for end users.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## How to test this PR locally
- Follow the `.buildkite/scripts/health-report-tests/README.md`
- CI test with this PR: https://buildkite.com/elastic/logstash-health-report-tests-pipeline/builds/58

## Related issues

- Continues https://github.com/elastic/ingest-dev/issues/3926

### CODE REVIEW

File: .buildkite/scripts/health-report-tests/main.py [hunk 1]
```diff
@@ -19,15 +19,15 @@ def __init__(self):
     def __enter__(self):
         print(f"Starting Logstash Health Report Integration test.")
         self.bootstrap = Bootstrap()
-        self.bootstrap.build_logstash()
+        # self.bootstrap.build_logstash()
```
Comment (mashhurs) [inline .buildkite/scripts/health-report-tests/main.py:22]:
```suggestion
        self.bootstrap.build_logstash()
```

File: .buildkite/scripts/health-report-tests/main.py [hunk 2]
```diff
@@ -19,15 +19,15 @@ def __init__(self):
...
 
         plugin_path = os.getcwd() + "/qa/support/logstash-integration-failure_injector/logstash-integration" \
                                     "-failure_injector-*.gem"
         matching_files = glob.glob(plugin_path)
         if len(matching_files) == 0:
             raise ValueError(f"Could not find logstash-integration-failure_injector plugin.")
 
-        self.bootstrap.install_plugin(matching_files[0])
+        # self.bootstrap.install_plugin(matching_files[0])
```
Comment (mashhurs) [inline .buildkite/scripts/health-report-tests/main.py:30]:
```suggestion
        self.bootstrap.install_plugin(matching_files[0])
```

File: .buildkite/scripts/health-report-tests/tests/slow-start.yaml [hunk 3]
```diff
@@ -8,7 +8,8 @@ config:
     pipeline.workers: 1
     pipeline.batch.size: 1
 conditions:
-  - full_start_required: false
+  full_start_required: false
+  wait_time_in_seconds: 5
```
Comment (mashhurs) [inline .buildkite/scripts/health-report-tests/tests/slow-start.yaml:12]: No need to wait for slow start scenario.
```suggestion
  wait_time_in_seconds: 0
```

File: .buildkite/scripts/health-report-tests/tests/multipipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            - impact_areas: ["pipeline_execution"]
+          details:
+            status:
+              state: "LOADING"
+        normally-terminated-pp-multipipeline:
+          status: "yellow"
+          symptom: "The pipeline is concerning; 1 area is impacted and 1 diagnosis is available"
+          diagnosis:
+            - cause: "pipeline has finished running because its inputs have been closed and events have been processed"
+            - action: "if you expect this pipeline to run indefinitely, you will need to configure its inputs to continue receiving or fetching events"
```
Comment (kaisecheng) [inline .buildkite/scripts/health-report-tests/tests/multipipeline.yaml:50]:
```suggestion
              action: "if you expect this pipeline to run indefinitely, you will need to configure its inputs to continue receiving or fetching events"
```

File: .buildkite/scripts/health-report-tests/tests/multipipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            - impact_areas: [ "pipeline_execution" ]
+          details:
+            status:
+              state: "FINISHED"
+        abnormally-terminated-pp-multipipeline:
+          status: "red"
+          symptom: "The pipeline is unhealthy; 1 area is impacted and 1 diagnosis is available"
+          diagnosis:
+            - cause: "pipeline is not running, likely because it has encountered an error"
+            - action: "view logs to determine the cause of abnormal pipeline shutdown"
```
Comment (kaisecheng) [inline .buildkite/scripts/health-report-tests/tests/multipipeline.yaml:61]:
```suggestion
              action: "view logs to determine the cause of abnormal pipeline shutdown"
```

File: .buildkite/scripts/health-report-tests/tests/multipipeline.yaml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+          impacts:
+            - description: "the pipeline is not currently processing"
+            - impact_areas: [ "pipeline_execution" ]
```
Comment (kaisecheng) [inline .buildkite/scripts/health-report-tests/tests/multipipeline.yaml:64]:
```suggestion
              impact_areas: [ "pipeline_execution" ]
```

File: .buildkite/scripts/health-report-tests/tests/multipipeline.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    pipelines:
+      status: "red"
+      symptom: "1 indicator is unhealthy (`abnormally-terminated-pp-multipipeline`) and 2 indicators are concerning (`slow-start-pp-multipipeline`, `normally-terminated-pp-multipipeline`)"
+      indicators:
+        slow-start-pp-multipipeline:
+          status: "yellow"
+          symptom: "The pipeline is concerning; 1 area is impacted and 1 diagnosis is available"
+          diagnosis:
+            - cause: "pipeline is loading"
+            - action: "if pipeline does not come up quickly, you may need to check the logs to see if it is stalled"
```
[the remark this answers is not in the document]
Comment (mashhurs) [reply .buildkite/scripts/health-report-tests/tests/multipipeline.yaml:39]: +1, ah good catch, this is I needed to standardize.

File: .buildkite/scripts/health-report-tests/tests/slow-start.yaml [hunk 8]
```diff
@@ -8,7 +8,8 @@ config:
     pipeline.workers: 1
     pipeline.batch.size: 1
 conditions:
-  - full_start_required: false
+  full_start_required: false
+  wait_time_in_seconds: 0
```
[the remark this answers is not in the document]
Comment (mashhurs) [reply .buildkite/scripts/health-report-tests/tests/slow-start.yaml:12]: It depends on scenario. For example, for this particular slow-start, we don't need to wait for pipeline to start. For rest, requires some time pipeline(s) to start or for the flow metrics, we need to waif 1m/5m to get the results.

**Review by kaisecheng — APPROVED** [review]
LGTM

**Pull request merged**